### PR TITLE
Publish to jfrog

### PR DESCRIPTION
**Pull request by tgdulia**

### CODE REVIEW

File: .github/workflows/publish-jfrog.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          cache: 'npm'
+          registry-url: 'https://egym.jfrog.io/egym/api/npm/npm-local/'
+          # Defaults to the user or organization that owns the workflow file
+          scope: '@egym'
+      - run: npm install
+      - run: npm run lint
+      - run: npm run build
+      - run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.EGYM_NPM_TOKEN }}
```
Comment (pedromedley) [inline .github/workflows/publish-jfrog.yml:26]: Will it work using the `secrets.EGYM_NPM_TOKEN` for authenticating against Artifactory?

I believe this contains the token to publish to npmjs, not artifactory.
In github actions there are the following env vars for artifactory:
```
ARTIFACTORY_USERNAME
ARTIFACTORY_PASSWORD
```

That is why we normally add the curl command to get .npmrc file form artifactory

File: .github/workflows/publish-jfrog.yml
Comment (pedromedley) [inline .github/workflows/publish-jfrog.yml:1]: For making a new release we also need to bump the version field in `package.json`

File: .github/workflows/publish-jfrog.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  release:
+    types: [published]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - uses: actions/checkout@v3
```
Comment (AlexKolomiytsev) [inline .github/workflows/publish-jfrog.yml:12]: please let's change actions to v4
actions/checkout@v4
actions/setup-node@v4

File: .github/workflows/publish-jfrog.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      # Setup .npmrc file to publish to GitHub Packages
```
Comment (AlexKolomiytsev) [inline .github/workflows/publish-jfrog.yml:13]: type in comment - `... to jfrog ...`

**Review by AlexKolomiytsev — APPROVED** [review]

**Review by pedromedley — APPROVED** [review]

**Pull request merged**